R/node-match.R

```diff
@@ -0,0 +1,454 @@
+#' Match patterns to a call
+#'
+#' @description
+#'
+#' `node_match()` is a handy way of checking whether a quoted call
+#' conforms to a pattern. The following rules apply to patterns:
```

Comment:
"The following rules apply to patterns:" -> "It uses the following rules"

Reply:
I think the new reader will become confused at this point if you do not add a more complete description of node_match before you move into pattern rules, e.g. 

"`node_match()` is a handy way of checking whether a quoted call conforms to a pattern. `node_match()` compares the call to a set of pattern ~ expression pairs. It evaluates the expression portion of the first pair whose pattern portion matches the call. 

`node_match()` uses the following rules to match call arguments to patterns..."